en/companion_computer/pixhawk_rpi.md

```diff
@@ -0,0 +1,259 @@
+PX4-Autopilot/Tools/setup/ubuntu.sh
+```
+
+Build it on the Pixhawk. On Linux, the default name of a USB connection is `/dev/ttyACM0`:
+
+```
+sudo chmod a+rw /dev/ttyACM0
+cd /PX4-Autopilot
+make px4_fmu-v6c_default upload
+```
+
+## Ubuntu setup on RPi
+
+The following steps are required to have the Ubuntu 22.04 installed on the RPi. In the next section, the ROS 2 is going to be installed, so it is important the choice for the correct version of Ubuntu.
+
+Prepare the Ubuntu 22.04's boot SD card following the official tutorial available in the Ubuntu website:
+
+https://ubuntu.com/tutorials/how-to-install-ubuntu-desktop-on-raspberry-pi-4#1-overview
+
+Next, connect the SD card in the RPi. Connect the mouse, keyboard, monitor and connect the RPi on 5v Power Supply (external source/charger).
+
+Turn on the RPi to boot on SD card and install Ubuntu. In the sequence, install raspi-config in the RPi's terminal:
+
+```
+sudo apt update
+sudo apt upgrade
+sudo apt-get install raspi-config 
+```
+
+Open the raspi-config:
+
+```
+sudo raspi-config
+```
+
+Go to the **Interface Option** and then click at **Serial Port**. Select **No** to disable serial login shell and select **Yes** to enable serial interface. Click **finish** and restart the RPi.
+
+In the sequence, in the RPi's terminal:
+
+```
+sudo nano /boot/firmware/config.txt
+```
+
+Include the following, after the last line:
+
+```
+enable_uart=1
+dtoverlay=disable-bt
+```
+
+Then, **ctrl+x**, **ctrl+y**, **Enter** and restart the RPi.
+
+To check if the serial port is available, in the RPi's terminal, confirm the list of serial devices by typing:
+
+```
+cd /
+ls /dev/ttyAMA0
+```
+
+On RPi, the default name of a RX/TX connection is `/dev/ttyAMA0`. The same serial port is also available as `/dev/serial0`.
+
+Next, install the mavproxy in the RPi's terminal:
+
+```
+sudo apt install python3-pip
+sudo pip3 install mavproxy
+sudo apt remove modemmanager
+```
+
+## MAVlink communication
+
+Connect the Pixhawk with the laptop via `USB` cable and change the following parameters in QGroundControl:
+
+```
+MAV_0_CONFIG = TELEM2
+XRCE_DDS_0_CFG = Disabled
+SER_TEL1_BAUD = 57600
+```
+
+Next, connect the Pixhawk `TELEM2` pins TX/RX/Ground on RPi correspondent pins. We are going to leave the `TELEM1` port for a setup with a Radio Controller.
+
+The connections listed below must be done:
+
+1. `TELEM2` TX -> RPi RXD (GPIO 15 - pin 10)
+2. `TELEM2` RX -> RPi TXD (GPIO 14 - pin 8)
+3. `TELEM2` GND -> RPi Ground (pin 6)
+
+It is important to identify the correct pins in the hardware. The image illustrates the `TELEM2` pins in the Pixhawk-6C Model.
+
+<img src="../../assets/companion_computer/pixhawk_rpi/pixhawk_pins.png" width="914" height="360" /> 
+
+Ps.: The pins are numbered according to the following:
+
+<img src="../../assets/companion_computer/pixhawk_rpi/pins_numbers.png" width="615" height="232" /> 
+
+The following image illustrates the correspondent pins at the RPi GPIO.
```

Comment:
Which RPi is this? I.e. I know there are a few RaPi's and these instructions will describe a particular config "mainly". But which one am I looking at?

I pretty much understand PX4 and Pixhawk. I know nothing much about RaPi, so this is where the holes are for me.

Reply:
> Which RPi is this? I.e. I know there are a few RaPi's and these instructions will describe a particular config "mainly". But which one am I looking at?
> 
> I pretty much understand PX4 and Pixhawk. I know nothing much about RaPi, so this is where the holes are for me.

Yes, I understand you. I only did not write specifically because, as you can see here:

https://www.raspberrypi.com/documentation/computers/raspberry-pi.html

"A 40-pin GPIO header is found on all current Raspberry Pi boards"

If you prefer we can list the models, no problem.

Reply:
My concern here is the instructions for setup. What I want to say is "connect the wires to this port on this output that will be present on all RaPis. That will always map to this particular `dev/Xxxx` value.". In that case we can treat all the boards as the same because as long as you connect to that port you specify the same values for your agent setup.

However my understanding was that different RaPis have different `dev/` values for different ports?

Reply:
In my understanding, the RX/TX pins of RPi map the communication to the dev/serial0 (same as ttyAMA0). Since we say that the connection must be in the pin 8 and pin 10 (RX/TX), and this is a standard IO pinout, it would result in a default approach independent of the RPi model.